### ISSUE TITLE
How long does server wait before closing an inactive connection?

### ISSUE THREAD

**Issue by zhiyong-ft** [ratchetphp/Ratchet#952]
When I was testing an embedded WS client with Ratchet, I noticed all connections were eventually closed by server after being inactive for certain period of time, ranging 1 ~ 2 hours, even if the client didn't gracefully close connections, due to power off or board being forcefully reset.

I briefly went through the code, seem not able to find the interval after which an idle connection will be closed. Can somebody advise.

Thanks.

**Comment by cboden** [comment]
Ratchet is not coded to close "idle" clients. If you're using the App class it will disconnect clients that are not able to communicate over a 30 second interval.

**Comment by zhiyong-ft** [comment]
That's interesting. I am not using the App class, and I commented out the following line that was recommended in your doc
```
//$wsServer->enableKeepAlive($gwLoop, 30);
```
Because it somehow interferes with the ping-pong we implemented in the embedded client.

However, I do have code on WS server that tracks marks connection as closed with a timestamp onClose. The timestamp when an idle connection last ping/message is usually 1 ~2.x hours after last ping/message from client.

I guess the only explanation is the idle connection is closed somewhere between the dead client and WS server.
